tests/std/tests/VSO_0157762_feature_test_macros/test.cpp

```diff
@@ -1073,19 +1073,29 @@ STATIC_ASSERT(__cpp_lib_apply == 201603L);
 #endif
 #endif
 
-#if CXX17_MODE
+
+#if CXX20_MODE
+#ifndef __cpp_lib_array_constexpr
+#error BOOM
+#elif __cpp_lib_array_constexpr != 201811L
+#error BOOM
+#else
+STATIC_ASSERT(__cpp_lib_array_constexpr == 201811L);
+#endif //__cpp_lib_array_constexpr != 201811L
+// ^^^ CXX20_MODE / CXX17_MODE vvv
+#elif CXX17_MODE
 #ifndef __cpp_lib_array_constexpr
 #error BOOM
 #elif __cpp_lib_array_constexpr != 201803L
 #error BOOM
 #else
 STATIC_ASSERT(__cpp_lib_array_constexpr == 201803L);
-#endif
-#else
+#endif ////__cpp_lib_array_constexpr != 201803L

```

Comment:
Unlike product code and most test code where we're very diligent about commenting preprocessor logic, in this test we've decided to give up and not comment any of these blocks - as they are extremely dense, it doesn't make them much easier to read. (I wish clang-format would indent preprocessor logic nicely.) I recommend removing all of these comments, although they may have helped during development, to be consistent with the rest of the test.

Reply:
I agree that consistency is more important here. And I also wish clang-format would indent this in a readable way.